packages/create-spectacle/src/cli.ts

```diff
@@ -12,49 +16,126 @@ import {
-        write(
-          chalk.whiteBright.bgRed.bold(' ! ') +
-            chalk.red.bold(' ' + message.replace('error: ', ''))
-        )
-    })
-    .requiredOption(
-      '-t, --type <type>',
-      'deck source type (choices: "tsx", "jsx", "mdx", "onepage")'
-    )
-    .requiredOption('-n, --name [name]', 'name of presentation')
-    .option(
-      '-l, --lang [lang]',
-      'language code for generated HTML document, default: en'
-    )
-    .option('-p, --port [port]', 'port for webpack dev server, default: 3000')
-    .parse(process.argv);
-
   let i = 0;
-  const { type, name, lang = 'en', port = 3000 } = program.opts<CLIOptions>();
+  let type = argv[ArgName.type] || argv['t'];
+  let name = argv['_']?.[0];
+  let lang = argv[ArgName.lang] || argv['l'] || 'en';
+  let port = argv[ArgName.port] || argv['p'] || 3000;
+
+  const isTryingToOverwrite = Boolean(name) && !isFolderNameAvailable(name);
+
+  /**
+   * If type/name not both provided via CLI flags, prompt for them.
+   */
+  if (
+    !(type && name && lang && (type === 'onepage' || port)) ||
+    isTryingToOverwrite
+  ) {
+    try {
+      const response = await prompts(
+        [
+          // Name prompt
+          {
+            type: 'text',
+            name: ArgName.name as string,
+            message: 'What is the name of the presentation?',
+            initial: name,
+            validate: async (val) => {
+              return val.length > 0 ? true : 'Name is required';
+            }
+          },
+          // If output directory already exists, prompt to overwrite
+          {
+            type: (val) => (isFolderNameAvailable(val) ? null : 'confirm'),
+            name: ArgName.overwrite as string,
+            message: (val) =>
+              `Target directory ${formatProjectDirName(
+                val
+              )} already exists. Overwrite and continue?`
+          },
+          // Check overwrite comes back false, we need to abort.
+          {
+            type: (_, answers) => {
+              if (answers?.[ArgName.overwrite] === false) {
+                throw new Error('❌ Operation cancelled');
+              }
+              return null;
+            },
+            name: 'overwriteAborter'
+          },
+          {
+            type: 'select',
+            name: ArgName.type as string,
+            message: 'What type of deck do you want to create?',
+            choices: DeckTypeOptions,
+            initial: (() => {
+              const ind = DeckTypeOptions.findIndex((o) => o.value === type);
+              return ind > -1 ? ind : 0;
+            })()
+          },
+          // Language prompt
+          {
+            type: 'text',
+            name: ArgName.lang as string,
+            message:
+              'What is the language code for the generated HTML document?',
+            initial: lang,
+            validate: async (val) => {
+              return val.length > 0 ? true : 'Language code is required';
```

Comment:
Should we require it? Or make `en` a default like the flags? Typically I've seen [en] default and pressing enter will go with that

Reply:
So the `initial: lang` bit will default it, so you can just hit Enter to select `en` by default. This validation is if the user Backspaces twice to try to use `""` as the input.